sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageCrc64Calculator.java

```diff
@@ -0,0 +1,2579 @@
+        0xc4d58efdc61b9ef6L,
+        0xa7e39e61e855bd45L,
+        0x97ad46f9dd1bf2f1L,
+        0x1a0abb01f853ee6bL,
+        0x3f0827c3348f8215L,
+        0x4eb68c4506134607L,
+        0x4a46f6de5df34e0aL,
+        0x2d855d6a1c57a8ddL,
+        0x8688da58e1115812L,
+        0x5232f417fc7c7300L,
+        0xa4080fb2e767d8daL,
+        0xd515a7e17693e562L,
+        0x1181f7c862e94226L,
+        0x9e23cd058204ca91L,
+        0x9b8992c57a0aed82L,
+        0xb2c0afb84609b6ffL,
+        0x2f7160553a5ea018L,
+        0x3cd378b5c99f2722L,
+        0x814054ad61a3b058L,
+        0xbf766189fce806d8L,
+        0x85a5e898ac49f86fL,
+        0x34830d11bc84f346L,
+        0x9644d95b173c8c1cL,
+        0x150401ac9ac759b1L,
+        0xebe1f7f46fb00ebaL,
+        0x8ee4ce0c2e2bd662L,
+        0x4000000000000000L,
+        0x2000000000000000L,
+        0x0800000000000000L, };
+
+    /**
+     * Computes the CRC64 checksum for the given byte array using the Azure Storage CRC64 polynomial.
+     * This method processes the input data in chunks of 32 bytes for efficiency and uses lookup tables
+     * to update the CRC values.
+     *
+     * @param src the byte array for which the CRC64 checksum is to be computed.
+     * @param uCrc the initial CRC value.
+     * @return the computed CRC64 checksum.
+     */
+    public static long compute(byte[] src, long uCrc) {
+        int pData = 0;
+        long uSize = src.length;
+        long uBytes, uStop;
+
+        uCrc = ~uCrc; // Flip all bits of uCrc
+
+        uStop = uSize - (uSize & 31); // Modulo of powers of 2 (2^i) can be performed with a bitwise operator
+
+        if (uStop >= 2 * 32) {
+            long uCrc0 = 0L;
+            long uCrc1 = 0L;
+            long uCrc2 = 0L;
+            long uCrc3 = 0L;
+
+            int pLast = pData + (int) uStop - 32;
+            uSize -= uStop;
+            uCrc0 = uCrc;
+
+            for (; pData < pLast; pData += 32) {
+                long b0, b1, b2, b3;
+
+                // Load and XOR data with CRC
+                b0 = ByteBuffer.wrap(src, pData, 8).order(ByteOrder.LITTLE_ENDIAN).getLong() ^ uCrc0;
+                b1 = ByteBuffer.wrap(src, pData + 8, 8).order(ByteOrder.LITTLE_ENDIAN).getLong() ^ uCrc1;
+                b2 = ByteBuffer.wrap(src, pData + 16, 8).order(ByteOrder.LITTLE_ENDIAN).getLong() ^ uCrc2;
+                b3 = ByteBuffer.wrap(src, pData + 24, 8).order(ByteOrder.LITTLE_ENDIAN).getLong() ^ uCrc3;
+
+                // Unsigned updates using tables and masking
+                uCrc0 = M_U32[7 * 256 + ((int) (b0 & 0xFF))];
+                b0 >>>= 8;
+                uCrc1 = M_U32[7 * 256 + ((int) (b1 & 0xFF))];
+                b1 >>>= 8;
+                uCrc2 = M_U32[7 * 256 + ((int) (b2 & 0xFF))];
+                b2 >>>= 8;
+                uCrc3 = M_U32[7 * 256 + ((int) (b3 & 0xFF))];
+                b3 >>>= 8;
+
+                // Repeat for remaining bytes
+                for (int i = 6; i >= 0; i--) {
```

Comment:
One thing I noticed looking over the .NET implementation: https://github.com/Azure/azure-sdk-for-net/blob/c85cc69742b4a5a325eb2f077cbdaf548e7e6362/sdk/storage/Azure.Storage.Common/src/Shared/StorageCrc64Calculator.cs#L840-L891 was that it did not use any for loops when doing these XORs. I'd prefer what we have for readability but curious if there was a specific reason the .NET implementation was done that way.

Reply:
we explicitly unrolled the loop. Whoever built this way back in the day for the service did the perf testing and wanted it that way.

Reply:
That's good to know. Thanks for confirming. @ibrahimrabab once we get the checksum working fully and some more performance benchmarking in place, let's try unrolling it to see if it makes a difference performance wise in the implementation.